cli/cli_app.go

```diff
@@ -715,6 +731,10 @@ func processHooks(hooks []config.Hook, terragruntOptions *options.TerragruntOpti
 				workingDir = *curHook.WorkingDir
 			}
 
+			rawActualLock, _ := sourceChangeLocks.LoadOrStore(workingDir, &sync.Mutex{})
+			actualLock := rawActualLock.(*sync.Mutex)
+			actualLock.Lock()
```

Comment:
Where's the unlock for this one?

Reply:
Oh, I see it way further down. Shouldn't it be in a `defer` right here to ensure it actually happens?

Reply:
hi, I will try to update this soon - most probably it should be extracted as separated function, or moved at beginning of hooks function